src/js/components/Beacon/BeaconQueryUi.js

```diff
@@ -0,0 +1,215 @@
+  useEffect(() => {
+    console.log('setting form initial values');
+    console.log({formInitialValues})
+    form.setFieldsValue(formInitialValues);
+  }, [config]);
+
+  const assemblyIdOptions = beaconAssemblyIds.map((assembly) => (
+    <Select.Option key={assembly} value={assembly}>
+      {assembly}
+    </Select.Option>
+  ));
+  const formInitialValues = { 'Assembly ID': beaconAssemblyIds.length == 1 && beaconAssemblyIds[0] };
+
+  // beacon request handling
+
+  const packageFilters = (values) => {
+
+    // ignore optional first filter when left blank 
+    if (filters.length == 1 && !values.filterId1) {
+      return []
+    }
+
+    return filters
+      .filter((f) => f.active)
+      .map((f) => ({
+        id: values[`filterId${f.index}`],
+        operator: '=',
+        value: values[`filterValue${f.index}`],
+      }));
+  };
+
+  const requestPayload = (query, payloadFilters) => ({
+    meta: { apiVersion: '2.0.0' },
+    query: { requestParameters: { g_variant: query }, filters: payloadFilters },
+    bento: { showSummaryStatitics: 'true' },
+  });
+
+  // following GA4GH recommendations, UI is one-based, but API is zero-based, "half-open"
+  // so to convert to zero-based, we only modify the start value 
+  // see eg https://genome-blog.soe.ucsc.edu/blog/2016/12/12/the-ucsc-genome-browser-coordinate-counting-systems/ 
+  const convertStartToZeroBased = (start) => Number(start)-1 
+
+  const packageBeaconJSON = (values) => {
+    let query = {};
+    const payloadFilters = packageFilters(values);
+    const hasVariantsQuery = (values && values['Chromosome']) || values['Variant start'] || values['Reference base(s)'];
+    if (hasVariantsQuery) {
+      query = {
+        referenceName: values['Chromosome'],
+        start: [convertStartToZeroBased(values['Variant start'])],
+        assemblyId: values['Assembly ID'],
+      };
+      if (values['Variant end']) {
+        query.end = [values['Variant end']];
+      }
+      if (values['Reference base(s)']) {
+        query.referenceBases = values['Reference base(s)'];
+      }
+      if (values['Alternate base(s)']) {
+        query.alternateBases = values['Alternate base(s)'];
+      }
+    }
+
+    return requestPayload(query, payloadFilters);
+  };
+
+  // form utils
+
+  const handleFinish = (formValues) => {
+    console.log('Received values of form: ');
+    console.log({ formValues: formValues });
+    const jsonPayload = packageBeaconJSON(formValues);
+
+    console.log({ queryPayload: jsonPayload });
+
+    dispatch(makeBeaconQuery(jsonPayload));
+  };
+
+  const handleClearForm = () => {
+    setFilters([STARTER_FILTER]);
+    form.resetFields();
+    form.setFieldsValue(formInitialValues);
+    launchEmptyQuery();
+  };
+
+  const SearchToolTip = ({ text }) => {
+    return (
+      <Tooltip title={text}>
+        <InfoCircleOutlined />
+      </Tooltip>
+    );
+  };
+  
+  // styles
+
+  const wrapperStyle = {
```

Comment:
move style constants outside function to avoid redefining every render

Reply:
Moved out. There's _so much_ style here and elsewhere that it would probably work better in a traditional css file, but it won't match the rest of the project.